src/shm_main.c

```diff
@@ -705,6 +742,14 @@ sr_shmmain_check_dirs(void)
     }
     free(dir_path);
 
+    /* connection lock dir */
+    sr_shmmain_get_conn_lockfile_path(SR_CONN_ID_NONE, &dir_path);
+    if ((err_info = sr_mkpath(dir_path, S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH))) {
```

Comment:
I personally like the octal permissions much more but since they are used everywhere in the code, it is also a formatting issue, so please conform to that. Moreover, you should be able to use some existing permissions in `common.h`.